### PR TITLE
Adds CI, specs, refactor implementation to meet spec

### PR DESCRIPTION
**Pull request by yaauie**
There is a lot going on here:

 - adds specs for behaviour
 - uses logstash-plugins/.ci to handle the bulk of ci bootstrapping running specs
 - refactor to adhere to specs and isolate functionality
 - touches up docs
 - adds Apache-2.0 `/LICENSE` (formerly only by reference in `*.gemspec`)
 - bumps version to 1.0.0 pending release

Specs output (`*` marked specs are omitted when native `Loggable#deprecation_logger` is present):
~~~
logstash_1  | LogStash::PluginMixins::DeprecationLoggerSupport
logstash_1  |   included into a class
logstash_1  |     that already includes Loggable
logstash_1  |       when instantiated
logstash_1  |         #deprecation_logger
logstash_1  |           behaves like memoized method
logstash_1  |             returns the same object from successive calls
logstash_1  |           the returned object
logstash_1  |             behaves like DeprecationLogger
logstash_1  |               is expected to respond to #deprecated
logstash_1  |               #deprecated
logstash_1  |                 when sent a single argument
logstash_1  |                   accepts a single string argument
logstash_1  |                   passes through to logger.warn *
logstash_1  |                 when sent two arguments
logstash_1  |                   accepts two arguments
logstash_1  |                   passes both arguments to logger.warn *
logstash_1  |         and class is a LogStash::Plugin
logstash_1  |           @deprecation_logger
logstash_1  |             behaves like DeprecationLogger
logstash_1  |               is expected to respond to #deprecated
logstash_1  |               #deprecated
logstash_1  |                 when sent a single argument
logstash_1  |                   accepts a single string argument
logstash_1  |                   passes through to logger.warn *
logstash_1  |                 when sent two arguments
logstash_1  |                   accepts two arguments
logstash_1  |                   passes both arguments to logger.warn *
logstash_1  |     that does not include Loggable
logstash_1  |       errors helpfully
logstash_1  |   extended into an object
logstash_1  |     that is Loggable
logstash_1  |       #deprecation_logger
logstash_1  |         behaves like memoized method
logstash_1  |           returns the same object from successive calls
logstash_1  |         the returned object
logstash_1  |           behaves like DeprecationLogger
logstash_1  |             is expected to respond to #deprecated
logstash_1  |             #deprecated
logstash_1  |               when sent a single argument
logstash_1  |                 accepts a single string argument
logstash_1  |                 passes through to logger.warn *
logstash_1  |               when sent two arguments
logstash_1  |                 accepts two arguments
logstash_1  |                 passes both arguments to logger.warn *
logstash_1  |     that is not Loggable
logstash_1  |       errors helpfully
logstash_1  |
logstash_1  | Finished in 1.01 seconds (files took 5.07 seconds to load)
logstash_1  | 19 examples, 0 failures
~~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Deprecation Logger Support Mixim
```
Comment (karenzone) [inline README.md]:
```suggestion
# Deprecation Logger Support Mixin
```

File: lib/logstash/plugin_mixins/deprecation_logger_support/legacy_init_adapter.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# encoding: utf-8
+
+module LogStash
+  module PluginMixins
+    module DeprecationLoggerSupport
+
+      ##
+      # The `LegacyInitAdapter` is used to hook into the intialization of
```
Comment (andsel) [inline lib/logstash/plugin_mixins/deprecation_logger_support/legacy_init_adapter.rb]: Err: intialization
Fix: initialization

File: lib/logstash/plugin_mixins/deprecation_logger_support/legacy_loggable_warn_adapter.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        # @api private
+        MEMOIZE_MUTEX = Mutex.new
+        private_constant :MEMOIZE_MUTEX
+
+        ##
+        # @return [DeprecationLogger]
+        def deprecation_logger
+          # threadsafe at-most-once memoize
+          # NOTE: the instance variable used here is _not_ a public part of the API.
+          @_deprecation_logger_legacy_adapter || MEMOIZE_MUTEX.synchronize do
```
Comment (andsel) [inline lib/logstash/plugin_mixins/deprecation_logger_support/legacy_loggable_warn_adapter.rb]: Good catch!

**Comment by andsel** [comment]
Nice peer I like the splitting of adapter operation and its initialization. Just typo fix, then 
LGTM

File: README.md (hunk 1 above)
Comment (yaauie) [reply README.md]: 🤦‍♂

**Pull request merged**